impala/_thrift_api.py

```diff
@@ -120,17 +122,19 @@ class ImpalaHttpClient(TTransportBase):
   MIN_REQUEST_SIZE_FOR_EXPECT = 1024
 
   def __init__(self, uri_or_host, port=None, path=None, cafile=None, cert_file=None,
-               key_file=None, ssl_context=None):
+               key_file=None, ssl_context=None, auth_cookie_name=None):
     """ImpalaHttpClient supports two different types of construction:
 
     ImpalaHttpClient(host, port, path) - deprecated
     ImpalaHttpClient(uri, [port=<n>, path=<s>, cafile=<filename>, cert_file=<filename>,
-        key_file=<filename>, ssl_context=<context>])
+        key_file=<filename>, ssl_context=<context>, auth_cookie_name=<cookiename>])
 
     Only the second supports https.  To properly authenticate against the server,
     provide the client's identity by specifying cert_file and key_file.  To properly
     authenticate the server, specify either cafile or ssl_context with a CA defined.
     NOTE: if both cafile and ssl_context are defined, ssl_context will override cafile.
+    auth_cookie_name is used to specify the name of the cookie used for cookie-based
```

Comment:
It seems inconvenient to require specifying this manually. Can we default it to 'impala.auth' at least? Or detect it automatically in setAuthCookie. E.g. if __auth_cookie_name is None, then check to see if "impala.auth' or 'hive.server2.auth' is present in the response and then use that?

Reply:
'auth_cookie_name' was renamed to 'auth_cookie_names' and now it takes a list of cookie names to look for.  The default value (specified in connect() API call) is ['impala.auth', 'hive.server2.auth'] so impyla will detect these by default.